src/fifelse.c

```diff
@@ -11,127 +11,180 @@ SEXP fifelseR(SEXP l, SEXP a, SEXP b, SEXP na) {
   const int64_t len0 = xlength(l);
   const int64_t len1 = xlength(a);
   const int64_t len2 = xlength(b);
+  const int64_t len3 = xlength(na);
   SEXPTYPE ta = TYPEOF(a);
   SEXPTYPE tb = TYPEOF(b);
+  SEXPTYPE tn = TYPEOF(na);
+  bool na_a = ta==LGLSXP && LOGICAL(a)[0]==NA_LOGICAL;
```

Comment:
@shrektan Great PR. Fully agree.
Just at this point, `a` could be length-0, and if so this `LOGICAL(a)[0]` would be an out-of-bound memory access.
Maybe the length checks later in this function could be moved up to be done first?